src/indexer/handlers/matchAmountUpdated.ts

```diff
@@ -1,24 +1,29 @@
-import { Indexer, JsonStorage, Event } from "chainsauce";
+import { Indexer, JsonStorage } from "chainsauce";
 import { convertToUSD } from "../../prices/index.js";
+import { Round } from "../types.js";
+import { MatchAmountUpdatedEvent } from "../events.js";
 
 export default async function (
   { chainId, storage: db }: Indexer<JsonStorage>,
-  event: Event
+  event: MatchAmountUpdatedEvent
 ) {
   const id = event.address;
   const matchAmount = event.args.newAmount.toString();
 
-  const round = await db.collection("rounds").findById(id);
+  const round = await db.collection<Round>("rounds").findById(id);
+
+  if (!round) {
```

Comment:
Same as above.

Reply:
This is an unexpected error, match amounts should always happen after a round is created. I have updated to throw an error.